### ISSUE TITLE
Refactor utililty functions out of generator into separate aio-utils npm package

### ISSUE THREAD

**Issue by icaraps** [adobe/generator-aio-app#130]
WDYT about moving all the utils functions into a separate npm package eg @adobe/aio-utils instead of generating the files in the templates ?

These ones to be precise:
* https://github.com/adobe/generator-aio-app/blob/master/generators/common-templates/utils.js
* https://github.com/adobe/generator-aio-app/blob/master/generators/add-web-assets/raw/templates/src/utils.js
* https://github.com/adobe/generator-aio-app/blob/master/generators/add-web-assets/exc-react/templates/src/utils.js

**Comment by meryllblanchet** [comment]
Thanks for the suggestion @icaraps !

@purplecabbage @rajarju wdyt?

**Comment by rajarju** [comment]
I think its a good Idea
It will also make it easier to replace UI code for devs and still use the utils if needed from the `aio-utils` package

**Comment by aiojbot** [comment]
JIRA issue created: https://jira.corp.adobe.com/browse/ACNA-1041

**Comment by purplecabbage** [comment]
https://git.corp.adobe.com/CNA/aio-exc-app 
Needs to be oss by @macdonst

**Comment by purplecabbage** [comment]
https://git.corp.adobe.com/OpenSourceAdvisoryBoard/opensource_submission_process/issues/1075

**Comment by shazron** [comment]
Done. https://github.com/adobe/aio-pkg-utils

**Comment by purplecabbage** [comment]
technically, the refactor to put this on npm, and use it in templates was not completed so I will reopen this with the intent of resolving it soon.

**Comment by purplecabbage** [comment]
A package was published to npm which includes import/requires that can be used in apps, however we decided that this would introduce unnecessary dependencies.  We decided it was simple and transparent enough to just drop these files into every new project and let devs modify them to suit their needs. This seems to be a better experience than hiding mystery code, even if it results in some duplication.
